blog/articles/event-drive-app-knative-eventing-kogito.md

```diff
@@ -0,0 +1,130 @@
+
+```yaml
+id: onboarding
+version: '1.0'
+name: Patient Onboarding Workflow
+states:
+- name: Onboard
+  type: event
+  start:
+    kind: default
+  onEvents:
+  - eventRefs:
+    - NewPatientEvent
+    actionMode: sequential
+    actions:
+    - functionRef:
+        refName: StoreNewPatient
+    - functionRef:
+        refName: AssignPatientToDoctor
+    - functionRef:
+        refName: SchedulePatientAppointment
+  end:
+    kind: default
+events:
+- name: NewPatientEvent
+  type: new.patient.events
+  source: "/hospital/entry"
+functions:
+- name: StoreNewPatient
+  operation: classpath:openapi.json#storeNewPatient
+- name: AssignPatientToDoctor
+  resource: classpath:openapi.json#assignPatientToDoctor
+- name: SchedulePatientAppointment
+  resource: classpath:openapi.json#schedulePatientAppointment
+```
+
+The above workflow definition is based on three parts. The first part is the “states” array which includes the workflow control-flow logic, in our case the “event” state which receives the new patient event and performs our three actions (service invocations).
+
+The second part is our event definitions (the “events” array), which are based on the CloudEvents specification, so there is a clear 1-1 mapping between how the events are defined in the workflows, and how they are represented with the CloudEvents format.
+
+The third part, the “functions” array, is the function definitions which give more information to our runtime about how to execute the needed services.
+
+Serverless Workflow specification is standards-based and leverages the OpenAPI specification to define details about restful executions of services. [In this gist](https://gist.github.com/ricardozanini/5ec66b4ddfbcf8ab40747b28b5f86333), you can see the OpenAPI file referenced by the functions in the above example.
+
+You can reference the [Serverless Workflow specification](https://github.com/serverlessworkflow/specification/blob/master/specification.md) for information about all language constructs that the specification provides.
+
+Note that the specification allows for both JSON and YAML workflow formats, and we used the YAML one in this example, but one based on JSON is considered equivalent and also parsable by the runtime.
+
+During compilation, the Kogito runtime will parse this YAML file and will generate Java code that represents this workflow definition. The generated code is based on the Quarkus framework. The outcome is an OpenAPI standard REST service that can be deployed anywhere in your architecture.
+
+![Kogito Runtime workflow parse process](images/kogito/kogito-process-workflow-file.png)*Kogito Runtime parsing flow*
+
+In this example, we also added the [Knative Kogito Eventing plugin](https://docs.jboss.org/kogito/release/latest/html_single/#con-knative-eventing_kogito-developing-process-services)  to the project, which means that it can accept Cloud Events through HTTP on the root path, for example:
```

Comment:
```suggestion
In this example, we have also added the [Knative Kogito Eventing plugin](https://docs.jboss.org/kogito/release/latest/html_single/#con-knative-eventing_kogito-developing-process-services) to the project, which means that it can accept CloudEvent objects through HTTP on the root path. For example:
```